shared/src/main/scala/kinesis4cats/logging/LogContext.scala

```diff
@@ -50,9 +53,21 @@ final class LogContext private (val context: Map[String, String]) {
   def addEncoded[A](
       key: String,
       a: A
-  )(implicit E: LogEncoder[A]): LogContext = new LogContext(
-    context + (key -> Option(a).fold("null")(x => E.encode(x)))
-  )
+  )(implicit E: LogEncoder[A]): LogContext =
+    add(Map(key -> Option(a).fold("null")(x => E.encode(x))))
+
+  def addRetryDetails(details: retry.RetryDetails): LogContext =
```

Comment:
I have a LogEncoder for RetryDetails that can be used instead of adding this function. Check out examples here:

https://github.com/etspaceman/kinesis4cats/blob/dbd4511b1dfc702e16da55f5a61fbc824abe9113/kcl/src/main/scala/kinesis4cats/kcl/RecordProcessor.scala#L124

https://github.com/etspaceman/kinesis4cats/blob/dbd4511b1dfc702e16da55f5a61fbc824abe9113/shared/src/main/scala/kinesis4cats/logging/instances/show.scala#L66-L91